### ISSUE TITLE
multi-gpu implementation

### ISSUE THREAD

**Issue by zpf0117b** [VincentStimper/normalizing-flows#39]
Does normflows support parallel computing on multi-GPU?

**Comment by VincentStimper** [comment]
Hi @zpf0117b,

Data parallelism, i.e. splitting the input batch over multiple devices, can be done with flow models build via this package in the same way as any other `torch.Module`. Simply use [`torch.nn.DataParallel`](https://pytorch.org/docs/stable/generated/torch.nn.DataParallel.html). An example is given [here](https://github.com/VincentStimper/resampled-base-flows/blob/master/experiments/train_glow.py).

Model parallelism is more difficult, but nut much different as with neural networks in torch I suppose 

Best regards,
Vincent
